spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/cassandra/CassandraMetricsBinder.java

```diff
@@ -0,0 +1,153 @@
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import com.codahale.metrics.Counting;
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.Timer;
+import com.datastax.oss.driver.api.core.CqlSession;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.binder.MeterBinder;
+
+import org.springframework.util.Assert;
+
+/**
+ * This is a {@link MeterBinder} that binds all available Cassandra DataStax driver
+ * metrics to Micrometer.
+ *
+ * @author Erik Merkle
+ * @since 2.3.0
+ */
+public class CassandraMetricsBinder implements MeterBinder {
+
+	/**
+	 * Prefix for all Driver configuration parameters.
+	 */
+	public static final String SESSION_PREFIX = "cassandra.driver.session";
+
+	/**
+	 * Description for micrometer metrics.
+	 */
+	public static final String DESCRIPTION = "Please see reference.conf for more information.";
+
+	private static final Map<String, Function<Snapshot, Number>> TIMER_SNAPSHOT_METRICS = new HashMap<>();
+
+	private static final Map<String, Function<Timer, Number>> TIMER_METRICS = new HashMap<>();
+
+	static {
+		TIMER_SNAPSHOT_METRICS.put("p999", Snapshot::get999thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p99", Snapshot::get99thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p98", Snapshot::get98thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p95", Snapshot::get95thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p75", Snapshot::get75thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("min", Snapshot::getMin);
+		TIMER_SNAPSHOT_METRICS.put("max", Snapshot::getMax);
+		TIMER_SNAPSHOT_METRICS.put("mean", Snapshot::getMean);
+		TIMER_SNAPSHOT_METRICS.put("median", Snapshot::getMedian);
+		TIMER_SNAPSHOT_METRICS.put("stdDev", Snapshot::getStdDev);
+	}
+
+	static {
+		TIMER_METRICS.put("mean_rate", Timer::getMeanRate);
+		TIMER_METRICS.put("m1_rate", Timer::getOneMinuteRate);
+		TIMER_METRICS.put("m5_rate", Timer::getFiveMinuteRate);
+		TIMER_METRICS.put("m15_rate", Timer::getFifteenMinuteRate);
+	}
+
+	private final Iterable<Tag> tags;
+
+	private final CqlSession session;
+
+	private CassandraMetricsSettings cassandraMetricsSettings;
+
+	public CassandraMetricsBinder(@NonNull CqlSession cqlSession,
+			@NonNull CassandraMetricsSettings cassandraMetricsSettings) {
+		this.cassandraMetricsSettings = cassandraMetricsSettings;
+		this.session = cqlSession;
+		this.tags = Tags.concat(Collections.emptyList(), "sessionName", cqlSession.getName());
+	}
+
+	@Override
+	public void bindTo(@NonNull MeterRegistry meterRegistry) {
+		Assert.isTrue(this.session.getMetrics().isPresent(), "Metrics on the CqlSession needs to be present");
+
+		this.session.getMetrics().get().getRegistry().getMetrics().forEach((name, m) -> {
```

Comment:
to validate if we are correctly registering different types of metrics, we should create an IT test that checks at least one metric of every type. Maybe you could follow this test:
https://github.com/riptano/java-driver-spring/blob/master/integration-tests/src/test/java/com/datastax/oss/spring/configuration/exampletests/CassandraMetricsMicrometerIT.java
It iterates over all metrics and validates that they are registered properly. I was able to catch some bugs when implementing original version using this test.

Reply:
I've added some more example metrics to the test [here](https://github.com/datastax/spring-boot/pull/2/files#diff-8991f02ba760e2c1590560601a62f5a1R58-R83).